File: Python/eight_threads_sequential.py
import random
import time

class Util:
	def selectionSort(self, list):
		n = len(list);
		for i in range (n - 1):
			min = i;
			j = i + 1;
			while(j < n):
				if(list[j] < list[min]):
					min = j;
				j = j + 1;
			temp = list[min];
			list[min] = list[i];
			list[i] = temp;
		return list;

	def createArray(self, start, end):
		list = [];
		for i in range(1000):
			temp = random.randint(start, end);
			list.append(temp);
		return list;

		
	def printArray(self, list):
		for i in list:
			print(i, end = " ");


	
def main():
	u1 = Util();
	
	list1 = u1.createArray(0, 9);
	print("Unsorted Array:");
	u1.printArray(list1);
	print();
	
	list2 = u1.createArray(10, 19);
	print("Unsorted Array:");
	u1.printArray(list2);
	print();
	
	list3 = u1.createArray(20, 29);
	print("Unsorted Array:");
	u1.printArray(list3);
	print();
	
	list4 = u1.createArray(30, 39);
	print("Unsorted Array:");
	u1.printArray(list4);
	print();
	
	list5 = u1.createArray(40, 49);
	print("Unsorted Array:");
	u1.printArray(list5);
	print();
	
	list6 = u1.createArray(50, 59);
	print("Unsorted Array:");
	u1.printArray(list6);
	print();
	
	list7 = u1.createArray(60, 69);
	print("Unsorted Array:");
	u1.printArray(list7);
	print();
	
	list8 = u1.createArray(70, 79);
	print("Unsorted Array:");
	u1.printArray(list8);
	print();
	
	
	start_time = time.time();
	list1 = u1.selectionSort(list1);
	print("Sorted Array:");
	u1.printArray(list1);
	print();
	
	list2 = u1.selectionSort(list2);
	print("Sorted Array:");
	u1.printArray(list2);
	print();
	end_time = time.time();
	
	list3 = u1.selectionSort(list3);
	print("Sorted Array:");
	u1.printArray(list3);
	print();
	
	list4 = u1.selectionSort(list4);
	print("Sorted Array:");
	u1.printArray(list4);
	print();
	
	list5 = u1.selectionSort(list5);
	print("Sorted Array:");
	u1.printArray(list5);
	print();
	
	list6 = u1.selectionSort(list6);
	print("Sorted Array:");
	u1.printArray(list6);
	print();
	end_time = time.time();
	
	list7 = u1.selectionSort(list7);
	print("Sorted Array:");
	u1.printArray(list7);
	print();
	
	list8 = u1.selectionSort(list8);
	print("Sorted Array:");
	u1.printArray(list8);
	print();
	
	end_time = time.time();
	
	print("--- Total Execution Time: %s seconds ---" % (end_time - start_time));
	
if(__name__ == '__main__'):
	main();

File: Java/EightThreadsSequential.java
// Total Execution Time of 'EightThreadsSequential': 0.4988 seconds
// Total Execution Time of 'EightThreadsThreaded': 0.2514 seconds
public class EightThreadsSequential
	{
	
	public static void main(String[] args){ 
	
	Handler t1 = new Handler();

	int arr1[] = t1.createArray(1, 9);
	System.out.println("Unsorted Array 1:");
	t1.printArray(arr1);
	
	int arr2[] = t1.createArray(11, 19);
	System.out.println("Unsorted Array 2:");
	t1.printArray(arr2);
	
	int arr3[] = t1.createArray(21, 29);
	System.out.println("Unsorted Array 3:");
	t1.printArray(arr3);
	
	int arr4[] = t1.createArray(31, 39);
	System.out.println("Unsorted Array 4:");
	t1.printArray(arr4);
	
	int arr5[] = t1.createArray(41, 49);
	System.out.println("Unsorted Array 5:");
	t1.printArray(arr5);
	
	int arr6[] = t1.createArray(51, 59);
	System.out.println("Unsorted Array 6:");
	t1.printArray(arr6);
	
	int arr7[] = t1.createArray(61, 69);
	System.out.println("Unsorted Array 7:");
	t1.printArray(arr7);
	
	int arr8[] = t1.createArray(71, 79);
	System.out.println("Unsorted Array 8:");
	t1.printArray(arr8);
	
	
		
	long startTime = System.nanoTime();
	arr1 = t1.selectionSort(arr1);
	arr2 = t1.selectionSort(arr2);
	arr3 = t1.selectionSort(arr3);
	arr4 = t1.selectionSort(arr4);
	arr5 = t1.selectionSort(arr5);
	arr6 = t1.selectionSort(arr6);
	arr7 = t1.selectionSort(arr7);
	arr8 = t1.selectionSort(arr8);
	long endTime   = System.nanoTime();	
	
	System.out.println("Sorted Array 1: ");
	t1.printArray(arr1);
	
	System.out.println("Sorted Array 2: ");
	t1.printArray(arr2);
	
	System.out.println("Sorted Array 3: ");
	t1.printArray(arr3);
	
	System.out.println("Sorted Array 4: ");
	t1.printArray(arr4);
	
	System.out.println("Sorted Array 5: ");
	t1.printArray(arr5);
	
	System.out.println("Sorted Array 6: ");
	t1.printArray(arr6);
	
	System.out.println("Sorted Array 7: ");
	t1.printArray(arr7);
	
	System.out.println("Sorted Array 8: ");
	t1.printArray(arr8);
	
	
	long totalTime = endTime - startTime;
	
	System.out.println("\n\nExecution Time: " + (double)totalTime / 1000000000);
	
 }  
}
